xgi/drawing/draw.py

```diff
@@ -985,3 +991,155 @@ def _update_lims(pos, ax):
+def draw_hypergraph_hull(
+        H, 
+        pos=None, 
+        ax=None, 
+        max_order=None, 
+        radius=0.05, 
+        alpha_filling=0.4, 
+        palette=None, 
+        node_size=60, 
+        node_color='tab:blue', 
+        node_shape='o', 
+        node_linewidths=1, 
+        node_edgecolors='black',
+        dyad_color='black'
+
+):
+    """Draw hypergraph with hyperedges of >3 elements displayed as convex hull
+
+    Parameters
+    ----------
+    H : Hypergraph
+    
+    pos : dict (default=None)
+        If passed, this dictionary of positions node_id:(x,y) is used for placing the nodes.
+        If None (default), use the `barycenter_spring_layout` to compute the positions.
+    
+    ax : matplotlib.pyplot.axes (default=None)
+    
+    max_order : int (default=None)
+        Maximum of hyperedges to plot. If None (default), plots all orders.
+    
+    radius : flot (default = 0.05)
+        radius of the convex hull in the vicinity of the nodes.
+    
+    alpha_filling : float (default=0.4)
+        The hyperedges transparency.
+    palette : None or array of colors (default=None)
+        Array of colors given by the hyperedges of different orders
+    
+    node_size : scalar or array (default=300)
+        Size of nodes.  If an array it must be the same length as nodelist.
+    
+    node_color : color or array of colors (default='tab:blue')
+        Node color. Can be a single color or a sequence of colors with the same
+        length as nodelist. Color can be string or rgb (or rgba) tuple of
+        floats from 0-1. If numeric values are specified they will be
+        mapped to colors using the cmap and vmin,vmax parameters. See
+        matplotlib.scatter for more details.
+    
+    node_shape : string (default='o')
+        The marker used for nodes, used in determining edge positioning.
+        Specification is as a `matplotlib.markers` marker, e.g. one of 'so^>v<dph8'.
+    
+    node_linewidths : None, scalar, sequence (default=1.0)
+        Line width of symbol border
+    
+    node_edgecolors : None, scalar, sequence (default = 'black')
+        Colors of node borders
+    
+    dyad_color : str, dict, iterable, or EdgeStat (default='black')
+        Color of the dyadic links.  If str, use the same color for all edges. If a dict, must
+        contain (edge_id: color_str) pairs.  If iterable, assume the colors are
+        specified in the same order as the edges are found in H.edges. If EdgeStat, use a colormap
+        (specified with dyad_color_cmap) associated to it
+
+    Returns
+    -------
+    ax : matplotlib.pyplot.axes
+
+    """
+    
+    if pos is None:
+        pos = barycenter_spring_layout(H)
+        
+    if ax is None:
+        ax = plt.gca()
+
+    ax.get_xaxis().set_ticks([])
+    ax.get_yaxis().set_ticks([])
+    ax.axis("off")
+    
+    if not max_order:
+        max_order = max_edge_order(H)
+    
+    if palette is None:
+        palette=sns.color_palette("deep", 6)
```

Comment:
I would probably pick a matplotlib palette to avoid having to add seaborn to the requirements just for this one

Reply:
Same as above, in my early attempts there were a lot of inconsistencies with the rest of xgi that I've corrected with the help of @maximelucas